packages/@aws-cdk/aws-cloudformation/lib/pipeline-actions.ts

```diff
@@ -92,10 +93,12 @@ export class PipelineExecuteChangeSetAction extends PipelineCloudFormationAction
       ChangeSetName: props.changeSetName,
     });
 
-    props.stage.pipelineRole.addToPolicy(new iam.PolicyStatement()
-      .addAction('cloudformation:ExecuteChangeSet')
-      .addResource(stackArnFromName(props.stackName))
-      .addCondition('StringEquals', { 'cloudformation:ChangeSetName': props.changeSetName }));
+    const sid = uuid(`PipelineExecuteChangeSetAction-${props.changeSetName}`);
```

Comment:
Why is it so important that this will be globally unique instead of just a well known SID that this construct uses and will not likely be defined by other users on the same policy? Feels like over doing it a little.

Reply:
I need to have an SID **for each ChangeSetName** because of that being presented in the `Condition`. And I don't know that I can guarantee the change set names will always be "Sid-friendly" (either in length or content). I just felt using an opaque hash would be easy.

I also considered using a well-known sting prefix, then a hash of the changeSet name... that brings legibility improvements to the policy, but makes it bigger (which is precisely what we want to avoid there).